libraries/cmd/collision.cpp

```diff
@@ -106,6 +106,12 @@ void Collision::shouldApplyForceAndDealDamage(Unit *other_unit) {
 
             // Units (ships) should calculate actual damage
         case Vega_UnitType::unit:
+            // Handle the "Nav 8" case
+            if ((other_unit->invisible & Unit::INVISUNIT)
```

Comment:
Shouldn't these conditions test be abstracted away in something like `other_unit->is_invisible()` ?

If I read this properly, does it mean I should rename all my units with "invisible" somewhere in their names ?

Reply:
Perhaps abstracting these conditions away would be a good idea. There's only two conditions though, IIRC, so I wasn't sure if it was worth it to create a whole new method. Especially since this would be the only place where it gets used.

Reply:
That's not the number of conditions that made me think it should be abstracted away (from my sight) ;-)

It's more related to my second question. It's the search for the "invisible" string in the unit name that I think is questionable.

So my question is in fact: why is that other_unit not having the invisible status and is using name tricks to cater for it ?

Reply:
So the search for the `"invisible"` text may not actually be required. That's the way I handled this issue previously, before I knew about the `invisible` property. For all I know, the `invisible` property by itself may be enough.

This needs to be tested. With PWCU. Unfortunately, PWCU's game assets are woefully out of date with respect to the latest Vega Strike Game Engine versions. So that will take some doing.

Reply:
> This needs to be tested. With PWCU. Unfortunately, PWCU's game assets are woefully out of date with respect to the latest Vega Strike Game Engine versions. So that will take some doing.

I think it would be easier to port the nav point to VS and add it to Cephid.

Reply:
I don't like any of this. It was bad code in the graphics. It's even worse here.
Better to check for something like `mass == 0` IMO.
But that might trigger some obscure bug somewhere.

Reply:
> > This needs to be tested. With PWCU. Unfortunately, PWCU's game assets are woefully out of date with respect to the latest Vega Strike Game Engine versions. So that will take some doing.
> 
> I think it would be easier to port the nav point to VS and add it to Cephid.

That makes sense. I'll try that.

Reply:
OK. So I took a look at the Nav_8 object and similar objects in PWCU, and imported Nav_8 into Cephid 17, as @royfalk suggested. I noticed that all the Nav_* objects had gravity == 0 and day == 240. The only other objects that had those same property values were jump points. It should not be possible to collide with any of the above, so I think those are the criteria I can use to determine whether an object can be collided with or not. No more string comparisons, which should be a lot better for performance.

Reply:
OK, guys, I think I have a solution. I implemented it as listed in the last comment above. It's efficient; it's encapsulated; and it seems to work reliably. Let me know what you guys think.

Reply:
To test it, I added the following to sectors/Crucible/Cephid_17.system, under "planet" Cephid 17 A:

```xml
	<planet name="Nav_8" file="invisible.png" alpha="ONE ONE" radius="256" gravity="0" x="4000" y="4000" day="240"/>
```

This was copied verbatim from the PWCU repo.